### PR TITLE
Update readme repo structure

### PR DESCRIPTION
**Pull request by Polichinel**
Updated readme to reflect the updated repo structure. I also expanded a bit on the notes accompanying the structure diagram. Lastly, since the root .gitignore seems to have gotten lost at some point, this PR also reintroduces this...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,157 +60,163 @@ Once the pipeline is initiated, you can monitor its execution using the Prefect
...
 |   ...
 |
 |-- ensembles/
-|   |--ensemble_001/ #similar to model dir, with a few differences
-|   |   |-- README.md 
-|   |   |-- requirements.txt   
+|   |-- exemplifying_ensemble/                      # Similar to model dir, with a few differences
+|   |   |-- README.md                               # Concise description of the ensemble and relevant details written in "plain language"
+|   |   |-- requirements.txt
+|   |   |-- main.py                                 # Orchestration script to run a deployed ensemble via Prefect on a monthly basis
```
Comment (sarakallis) [inline README.md:150]: Same comment as above re: Prefect

File: README.md [hunk 2]
```diff
@@ -60,157 +60,163 @@ Once the pipeline is initiated, you can monitor its execution using the Prefect
...
 |   |   |
-|   |   |-- configs/
-|   |   |   |-- config_sweep.py 
-|   |   |   |-- config_hyperparameters.py
+|   |   |-- configs/                                # All ensemble specific config files
+|   |   |   |-- config_model.py                     # Contains ensemble architecture, name, target variable, and level of analysis
+|   |   |   |-- config_hyperparameters.py           # If applicable, specifies the finalized hyperparameters of the ensemble (W&B specific)
+|   |   |   |-- config_sweep.py                     # If applicable, specifies the hyperparameter sweeps during experimentation phases (W&B specific)
+|   |   |   |-- config_deployment.py                # Status of the ensemble regarding its lifecycle. E.g., production, shadow mode, or a baseline model
```
Comment (sarakallis) [inline README.md:156]: Same as above

File: README.md [hunk 3]
```diff
@@ -60,157 +60,163 @@ Once the pipeline is initiated, you can monitor its execution using the Prefect
...
+|
+|-- documentation/                                  # All relevant documentation (wiki to come)
 |
-|-- prefect/
-|    |--scripts.py # orchestration
+|-- common_utils/                                   # Functions and classes used across multiple (but not necessarily all) models/ensembles  
+|       |-- stepshifter.py                          # Updated stepshifter function
+|       |-- set_paths.py                            # Sets all paths for imports, data, utils etc. Machine invariant.
+|       |-- get_data.py                             # General function that takes the general get_partition and a model specific config_feature_set.py to fetch model specific data
+|       |-- get_partition.py                        # Get data partitions for splits pertaining to validation, testing, and forecasting
```
Comment (sarakallis) [inline README.md:213]: Shouldn't this be in a config file instead, since we are just changing parameters?

**Review by sarakallis — CHANGES_REQUESTED** [review]
Looks good, but my 2 desired changes would be to 
1. add config_deployment to config_model
2. make common_utils/get_partition into a config file (since my understanding is that we are defining, not generating per se)

File: README.md [hunk 4]
```diff
@@ -60,157 +60,163 @@ Once the pipeline is initiated, you can monitor its execution using the Prefect
...
-|-- models/
-|   |-- model_001/ # should follow the naming convention adjective_noun
+|-- models/                                         # Parent directory for all individual models (see documentation for definition of "model")
+|   |-- exemplifying_model/                         # Each individual model subdirectory should follow the naming convention adjective_noun
 |   |   |
-|   |   |-- README.md
-|   |   |-- requirements.txt
+|   |   |-- README.md                               # Concise description of the model and relevant details written in "plain language"
+|   |   |-- requirements.txt                        # Python version and libraries - should rarely deviate from a standard well-maintained VIEWS_env.
+|   |   |-- main.py                                 # Orchestration script to run a deployed model via Prefect on a monthly basis
```
[the remark this answers is not in the document]
Comment (Polichinel) [reply README.md]: Noted - I've clarified the description.

File: README.md [hunk 5]
```diff
@@ -60,157 +60,163 @@ Once the pipeline is initiated, you can monitor its execution using the Prefect
...
-|   |   |-- configs/ # ...
-|   |   |   |-- config_sweep.py # hyperparameters for test_sweep.py (WandB sweep)
-|   |   |   |-- config_hyperparameters.py # hyperparameters train_model.py
-|   |   |   |-- config_partitioner.py OR config_common # calibration, test, forecasting partitions
-
+|   |   |-- configs/                                # All model specific config files
+|   |   |   |-- config_model.py                     # Contains model architecture, name, target variable, and level of analysis
+|   |   |   |-- config_hyperparameters.py           # Specifies the finalized hyperparameters used for training the deployed model (W&B specific)
+|   |   |   |-- config_sweep.py                     # Configurations for hyperparameter sweeps during experimentation phases (W&B specific)
+|   |   |   |-- config_feature_set.py               # Defines the features to be pulled from the views - basically the queryset
```
[the remark this answers is not in the document]
Comment (Polichinel) [reply README.md:86]: No strong feelings here so done

File: README.md (hunk 1 above)
Comment (Polichinel) [reply README.md]: clarified description

File: README.md (hunk 2 above)
Comment (Polichinel) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (Polichinel) [reply README.md]: Jeps - actually did that at one point bc it made sense, but then realized that was not what we agreed upon so changed it to common_utills. However, I've now made a common_config dir in root and put it there - as we both seems to agree that make more sense

**Review by sarakallis — APPROVED** [review]

**Pull request merged**